### DISCUSSION TITLE
Include our own custom components?

### DISCUSSION THREAD

**Discussion by grijalva10** [Budibase/budibase#1390] [Help]
How would I go about including my own svelte components to the designer?

**Comment by mjashanks** [comment]
This is currently not possible - we have not started work on it yet. However, we intend to have this feature available in the next 3-6 months.

**Reply by grijalva10** [reply]
Awesome. Thanks for the quick response.

**Reply by justinr1234** [reply]
@mjashanks is there any new info here? Thanks!

**Reply by mjashanks** [reply]
@justinr1234  There is no new info here. We keep pushing it out in favour of other things. Can't give you a date for it, but people ask me all the time - so it's still on the radar. Sorry that I can't be more definite.

**Reply by justinr1234** [reply]
@mjashanks Is there any idea or prelim spec of what’s needed? Is it something the community could assist with?

**Comment by JohnRSim** [comment]
@mjashanks can we use https://github.com/Budibase/bbui to build custom components for now? 
What's the roadmap for custom components development.

Thanks

**Reply by mjashanks** [reply]
Unfortunately not, this kept getting pushed out by other things. I can't really tell you a date at this stage.

Is there anything in particular that you would like to build, that you can share?

**Reply by JohnRSim** [reply]
- Infinite scroll news feed widget with personalised content (similar to instagram)
- Chatbot Widget
- Chat Widget (similar to whatsapp with group discussion capabilities integrated with sendbird)
- Social Post widget - allow users to post video, images, text - (similar to instagram/twitter)
- Custom video player with streaming support
- Predesigned SaaS widgets for HR, payroll etc that users can drop in that are preconfigured UI/UX.

**Reply by JohnRSim** [reply]
![image](https://user-images.githubusercontent.com/364208/136181360-0f3de616-0334-4239-932b-b9507669a41b.png)
Something like this allows users to scroll through content quickly.

**Reply by JohnRSim** [reply]
Any progress on this or still pushed back - thanks

**Reply by mjashanks** [reply]
Hey John - it's still pushed back. It won't be this quarter, **might** be the first half of 2022.

**Reply by miconx** [reply]
> Hey John - it's still pushed back. It won't be this quarter, **might** be the first half of 2022.

would like to use/buy budibase for building internal apps for our  company
but we need to use widgets that are not available in budibase
is there any manual/tutorial where we could learn howto do this?
we are regulated to use on premise software so we are not allowed to use cloudbased products

**Comment by bradleyandrew** [comment]
Hello,

I hope this message finds you well!

I would also like to chime in here, I would love a Slider UI Element, something that can manipulate a Double Value in the Database with a horizontal slider. Vertical slider would also be a nice option. If users could import Custom UI that would be ideal but I did want to mention a slider just encase it's something you were thinking about building internally.

Thanks!

**Reply by gaa23** [reply]
Would love this too. IDK what this is, but it looks like a slider component from the github: https://github.com/Budibase/bbui/blob/master/src/Form/Slider.svelte
